gnat2goto/driver/tree_walk.adb

```diff
@@ -4083,12 +4095,15 @@ package body Tree_Walk is
    function Do_Selected_Component (N : Node_Id) return Irep is
       Root           : constant Irep := Do_Expression (Prefix (N));
       Component      : constant Entity_Id := Entity (Selector_Name (N));
+      Orig_Component : constant Entity_Id :=
```

Comment:
Can you add a short explanation for what an original record component is and why we need it here?

Reply:
```
struct Foo { int a; };
struct Foo bar;
bar.a = 5;
```
So the `Unique_Name` of the struct-component in declaration is `Foo__a`. But when parsing the assignment the component is that of the entity `bar`, thus it's unique-name is `bar__a`: which is not present in `Foo`. That's why we have to use the component of the original-record to get the right name. (I'll add this as a comment.)